### DISCUSSION TITLE
setLabel is not generating Label tab on UI

### DISCUSSION THREAD

**Discussion by promodsingh** [primefaces/community#3163] [PrimeFaces] [answered]
Hello,

I am programmatically creating components but setLabel() method on UI components like InputText is not creating Label tag on UI. Can you help to understand whether this method works or I need to explicitly use OutputLabel.

Thanks

**Comment by tandraschko** [comment]
you need the outputlabel

**Comment by promodsingh** [comment]
Ok, Actually I am looking for something like ADF or trinidad components, we set label it generates the label automatically

**Reply by tandraschko** [reply, marked as answer]
we dont have something like that
for us it was always better to have this in seperate components

**Comment by promodsingh** [comment]
Ok, Thanks @tandraschko.
